strawberry/federation/schema_directives.py

```diff
@@ -23,3 +28,42 @@ class Provides:
 @schema_directive(locations=[Location.OBJECT, Location.INTERFACE], name="key")
 class Key:
     fields: FieldSet
+    resolvable: bool = True
+
+
+# @schema_directive(locations=[Location.SCHEMA], name="link")
+# class Link: # todo: address this in some other PR ... it needs to be handled differently than these other directives.
+#     url: str
+#     link_as: str # "as" collides with Python reserved word "as"
+#     link_for: LinkPurpose # "for" collides with Python reserved word "for"
+#     link_import: LinkImport  # "import" is reserved word
+
+
+@schema_directive(
+    locations=[Location.FIELD_DEFINITION, Location.OBJECT], name="shareable"
+)
+class Shareable:
+    ...
+
+
+@schema_directive(locations=[Location.FIELD_DEFINITION], name="tag")
```

Comment:
this is supported on all these:

```
FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
```

source: https://cs.github.com/apollographql/federation/blob/7331cbdb824af20f70ca500f1826859c92edda49/federation-integration-testsuite-js/src/fixtures/reviews.ts?q=directive+%40tag#L9-L19

Reply:
resolved